### ISSUE TITLE
Fix CMake 3.9 warnings

### ISSUE THREAD

**Issue by jcfr** [SuperElastix/elastix#21]
Configuring the extension using the latest CMake release reports the following warnings:

```
CMake Deprecation Warning at src/CMakeLists.txt:7 (cmake_policy):
  The OLD behavior for policy CMP0033 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


CMake Deprecation Warning at src/CMakeLists.txt:10 (cmake_policy):
  The OLD behavior for policy CMP0007 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


CMake Warning (dev) at /home/jcfr/Software/cmake-3.9.0-Linux-x86_64/share/cmake-3.9/Modules/FindOpenMP.cmake:200 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "c" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /home/jcfr/Software/cmake-3.9.0-Linux-x86_64/share/cmake-3.9/Modules/FindOpenMP.cmake:324 (_OPENMP_GET_FLAGS)
  src/CMakeLists.txt:95 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by mstaring** [comment]
@N-Dekker would you have time to look at this?

**Comment by N-Dekker** [comment]
For the record, I can reproduce the issue (warnings for policy CMP0033 and CMP0007) on CMake 3.10.2 (for generator Visual Studio 2015 Win64).

**Comment by N-Dekker** [comment]
"SET CMP0033 OLD" allows using export_library_dependencies(): https://cmake.org/cmake/help/v3.10/policy/CMP0033.html
"SET CMP0007 OLD" means that empty list items (for example the item between 'b' and 'c' in "a;b;;c") are ignored: https://cmake.org/cmake/help/v3.10/policy/CMP0007.html

The way these policies are set is also related to cmake_minimum_required, with is VERSION 2.8, in Elastix. SuperElastix does cmake_minimum_required( VERSION 3.5.1 ). ITK even does cmake_minimum_required(VERSION 3.9.5 FATAL_ERROR). Are there any plans for Elastix to move forward with cmake_minimum_required?

**Comment by jcfr** [comment]
> Are there any plans for Elastix to move forward with cmake_minimum_required?

👍  That would be ideal.

**Comment by mstaring** [comment]
@jcfr Would moving to cmake version 3.5.1 work for you?

**Comment by jcfr** [comment]
> Would moving to cmake version 3.5.1 work for you?

👍  Definitively. And thanks for checking

ITK already trunk moved to 3.9.5 and Slicer will follow shortly.

See https://discourse.slicer.org/t/updating-minimum-required-version-of-cmake-to-3-9-5/1873

**Comment by mstaring** [comment]
@N-Dekker can you update me on the status of this issue?

**Comment by N-Dekker** [comment]
@mstaring So far, I haven't figured out how to replace the export_library_dependencies call from the CMakeLists. Which was needed in order to upgrade cmake_minimum_required. Is there an extra reason to revisit this issue now. Trying to fix it appeared quite time-consuming!

**Comment by mstaring** [comment]
Well, the importance is based on other packages that include elastix (in this case slicer) are moving forward with their cmake versions. It would be nice to modernize. Let's discuss this on the next sprint.

**Comment by N-Dekker** [comment]
Here is where I got stuck the last time, trying to link SuperElastix to elastix, after having upgraded cmake_minimum_required (at elastix branch https://github.com/SuperElastix/elastix/tree/ELX-21-Fix-CMake-warnings) and removed export_library_dependencies: https://my.cdash.org/viewBuildError.php?buildid=1374121 

> undefined reference to `typeinfo for itk::SingleValuedCostFunction'
> undefined reference to `itk::ProgressAccumulator::ProgressAccumulator()'

etc. 

I tried to link SuperElastix branch https://github.com/SuperElastix/SuperElastix/tree/SELX-171-Check-elastix-issue-21

**Comment by jiwoosong** [comment]
Is this problem resolved?

**Comment by N-Dekker** [comment]
@jiwoosong This problem is addressed by commit https://github.com/SuperElastix/elastix/commit/8707a4aaad330ae4554f55e505889ef61d6a84dc "COMP: Fix CMake 3.x warnings policy CMP0033 and CMP0007 (issue #21) ", 12 September 2020 (included with elastix version 5.0.1 and the develop branch). Does it work for you?
